api/src/lib.rs

```diff
@@ -2273,6 +2280,209 @@ ffi_export!(
+            "updatedSigningKeys",
+            "com/horizen/schnorrnative/SchnorrPublicKey",
+        );
+        let _schnorr_updated_master_keys_list = parse_jobject_array_from_jobject(
+            &_env,
+            _keys_signatures_list,
+            "updatedMasterKeys",
+            "com/horizen/schnorrnative/SchnorrPublicKey",
+        );
+
+        let _updated_signing_keys_sk_signatures = parse_jobject_array_from_jobject(
+            &_env,
+            _keys_signatures_list,
+            "updatedSigningKeysSkSignatures",
+            "com/horizen/schnorrnative/SchnorrSignature",
+        );
+        let _updated_signing_keys_mk_signatures = parse_jobject_array_from_jobject(
+            &_env,
+            _keys_signatures_list,
+            "updatedSigningKeysMkSignatures",
+            "com/horizen/schnorrnative/SchnorrSignature",
+        );
+        let _updated_master_keys_sk_signatures = parse_jobject_array_from_jobject(
+            &_env,
+            _keys_signatures_list,
+            "updatedMasterKeysSkSignatures",
+            "com/horizen/schnorrnative/SchnorrSignature",
+        );
+        let _updated_master_keys_mk_signatures = parse_jobject_array_from_jobject(
+            &_env,
+            _keys_signatures_list,
+            "updatedMasterKeysMkSignatures",
+            "com/horizen/schnorrnative/SchnorrSignature",
+        );
+
+        let withdrawal_certificate = parse_wcert(_env, _withdrawal_certificate);
+        let prev_withdrawal_certificate =
+            cast_joption_to_rust_option(&_env, _prev_withdrawal_certificate)
+                .map(|cert_object| parse_wcert(_env, cert_object));
+        let genesis_key_root_hash = convert_field_element(&_env, _genesis_key_root_hash);
+
+        let signing_keys = extract_public_key(&_env, _schnorr_signing_keys_list);
+        let master_keys = extract_public_key(&_env, _schnorr_master_keys_list);
+        let updated_signing_keys = extract_public_key(&_env, _schnorr_updated_signing_keys_list);
+        let updated_master_keys = extract_public_key(&_env, _schnorr_updated_master_keys_list);
+
+        let updated_signing_keys_sk_signatures: Vec<_> =
+            JObjectArrayIter::new(&_env, _updated_signing_keys_sk_signatures)
+                .map(|s| *convert_signature(&_env, s))
+                .collect();
+        let updated_signing_keys_mk_signatures: Vec<_> =
+            JObjectArrayIter::new(&_env, _updated_signing_keys_mk_signatures)
+                .map(|s| *convert_signature(&_env, s))
+                .collect();
+        let updated_master_keys_sk_signatures: Vec<_> =
+            JObjectArrayIter::new(&_env, _updated_master_keys_sk_signatures)
+                .map(|s| *convert_signature(&_env, s))
+                .collect();
+        let updated_master_keys_mk_signatures: Vec<_> =
+            JObjectArrayIter::new(&_env, _updated_master_keys_mk_signatures)
+                .map(|s| *convert_signature(&_env, s))
+                .collect();
```

Comment:
We need to call here `convert_optional_signature` as Java side empty SchnorrSignature instances will be initialized in case of no key change.
We didn't catch it as the test, Java-side, doesn't stress key changes.
I'm not saying we should write exhaustive tests also Java side, as we have them in Rust and in the SDK, but let's try to stress a bit all of the components of the circuit.

Reply:
Updated the parsing, but we still need to fix the Java test